sql-plugin/src/main/scala/com/nvidia/spark/rapids/AlluxioUtils.scala

```diff
@@ -657,4 +832,46 @@ object AlluxioUtils extends Logging with Arm {
     }
     (location, mapIfReplacedPaths)
   }
+
+  def isLargeTable(map: mutable.Map[String, mutable.Set[String]],
+      bucket: String, path: String): Boolean = {
+    val largeSetOption = map.get(bucket)
+    largeSetOption.exists { set =>
+      set.exists { largeTable =>
+        path.startsWith(largeTable)
+      }
+    }
+  }
+
+  def directlyReadLargeTableFromS3(
+      paths: Seq[Path], conf: RapidsConf, relation: HadoopFsRelation): Boolean = {
+    if (conf.enableAlluxioSlowDisk) {
+      // if enabled directly reading from s3 for large files
+      val hadoopConf = relation.sparkSession.sparkContext.hadoopConfiguration
+      val runtimeConf = relation.sparkSession.conf
+      initAlluxioInfo(conf, hadoopConf, runtimeConf)
+      directlyReadLargeTableFromS3(largeTableMap, paths)
+    } else {
+      false
+    }
+  }
+
+  def directlyReadLargeTableFromS3(map: mutable.Map[String, mutable.Set[String]],
```

Comment:
I would prefer function name to be like shouldReadDirectly... also rename the input parameter name to be more useful then just map... like largeTableMap or something.

Reply:
Done.